src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -347,6 +347,8 @@ static inline int MPIDI_OFI_do_put(const void *origin_addr,
     MPIDI_Datatype_check_contig_size_lb(origin_datatype, origin_count, origin_contig, origin_bytes,
                                         origin_true_lb);
 
+    MPIDI_CH4U_RMA_OP_CHECK_SYNC(target_rank, win);
+
```

Comment:
The `MPIDI_CH4U_RMA_OP_CHECK_SYNC ` check should be done before `if (unlikely(target_rank == MPI_PROC_NULL)) goto null_op_exit;`. For example, it is invalid if the user calls `MPI_PUT(PROC_NULL...)` before a `MPI_WIN_LOCK_ALL`.

Reply:
See the location in `MPIDI_OFI_do_accumulate`.

Reply:
Yes that's a mistake. Fixed.
Also, I just noticed that some `if(target_rank == MPI_PROC_NULL)` does not have unlikely(). So I add another commit for this fix.